src/main/java/de/tum/in/www1/artemis/service/connectors/jenkins/JenkinsUserManagementService.java

```diff
@@ -128,6 +128,28 @@ public void createUser(User user) throws ContinuousIntegrationException {
         return new HttpEntity<>(formData, headers);
     }
 
+    /**
+     * Checks if the Artemis you can be created in Jenkins. The
+     * function first checks if the user exists and if it doesn't
+     * attempts to create it. If the creation is successful, the
+     * Jenkins user is deleted and a boolean value is returned.
+     *
+     * @param user The user to check
+     * @return whether the user can be created in Jenkins or not
+     */
+    @Override
+    public boolean canCreateUser(User user) {
+        try {
+            createUser(user);
```

Comment:
To be honest I am not a big fan of such a logic, because something like this should be avoided in my opinion. Can we not find out differently if a user can be created?

In any case, I suggest to move the deleteVcsUser into the finally block.